### ISSUE TITLE
Automatically translate C++ into IPR and back

### ISSUE THREAD

**Issue by spaanse** [GabrielDosReis/ipr#137]
In [the paper](https://stroustrup.com/macis09.pdf) is is mentioned that

>  We generate IPR from two compiler frontends[6, 9]
> ...
> [6] The Edison Design Group.http://www.edg.com/.
> [9] GNU Compiler Collection.http://gcc.gnu.org/.

The two links don't specify how to generate IPR, they are just links to the frontends.
Is there anywhere I can find how to transform C++ code automatically into IPR and back?

**Comment by GabrielDosReis** [comment]
> The two links don't specify how to generate IPR, they are just links to the frontends.

Those were the original projects we worked on, hooking IPR into GCC and EDG frontends.  The GCC option for dumping a translation unit into a format (on-disk) that was eventually read into IPR was eventually removed from recent GCC.  The integration of the IPR to EDG was never contributed back to EDG so it is not part of that front-end.

There is a recent effort within Microsoft to build IPR from the Clang AST, and build IPR from the MSVC compiler.  More on the later in coming months.  See also my [keynote talk](https://www.youtube.com/watch?v=j4du4LNsLiI) at Meeting C+ 2020.

You might also be interested in [this community request](https://developercommunity2.visualstudio.com/t/Add-support-to-clexe-to-create-ifc-/1293280) - upvote it if it would be useful for your development scenarios

**Comment by herrgahr** [comment]
Right, this answers to C++ -> IPR question. I'm perusing the issues because I am interested in the IPR -> C++ direction, just like OP.

I'm working on codegen tools and I think it might be beneficial to build IPR and then translate that IPR to C++.

AFAICT there is nothing like this hiding in the src tree.

Are there any plans, implementations or prototypes?
